.github/workflows/build_and_test.yaml

```diff
@@ -1,10 +1,16 @@
 name: Build and Test
 on:
   push:
+    branches:
+      - "main"
+      - "release-v*"
```

Comment:
`release/*` for consistency w/envoy

Reply:
we've already started with `release-v0.1.0` (https://github.com/envoyproxy/gateway/tree/release-v0.1.0)
@danehans any idea why we picked this convention ?

Reply:
we can rename branches anyway

Reply:
@arkodg I used Envoy as the model for Gateway release branch naming, substituting `/` with `-` since `-` is more widely used.

Reply:
cool thanks, I'll leave the PR as is, since the naming was intentional, we can change it, if we change the branch naming in the future ?